Build for debian stretch Melodic is failing on ROS buildfarm (cmake and pthread)

http://build.ros.org/job/Mbin_ds_dS64__neonavigation__debian_stretch_amd64__binary/
![image](https://user-images.githubusercontent.com/8390204/80917718-de76db00-8d9b-11ea-8ec2-215332dc2786.png)
```
08:12:37 	cd /tmp/binarydeb/ros-melodic-map-organizer-0.8.4
08:12:37 	cd obj-x86_64-linux-gnu
08:12:37 	"tail -v -n +0 CMakeFiles/CMakeError.log"
08:12:37 ==> CMakeFiles/CMakeError.log <==
08:12:37 Determining if the pthread_create exist failed with the following output:
08:12:37 Change Dir: /tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp
08:12:37 
08:12:37 Run Build Command:"/usr/bin/make" "cmTC_4cc55/fast"
08:12:37 make[2]: Entering directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 /usr/bin/make -f CMakeFiles/cmTC_4cc55.dir/build.make CMakeFiles/cmTC_4cc55.dir/build
08:12:37 make[3]: Entering directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 Building C object CMakeFiles/cmTC_4cc55.dir/CheckSymbolExists.c.o
08:12:37 /usr/bin/cc    -g -O2 -fdebug-prefix-map=/tmp/binarydeb/ros-melodic-map-organizer-0.8.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2    -o CMakeFiles/cmTC_4cc55.dir/CheckSymbolExists.c.o   -c /tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/CheckSymbolExists.c
08:12:37 Linking C executable cmTC_4cc55
08:12:37 /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_4cc55.dir/link.txt --verbose=1
08:12:37 /usr/bin/cc  -g -O2 -fdebug-prefix-map=/tmp/binarydeb/ros-melodic-map-organizer-0.8.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2     CMakeFiles/cmTC_4cc55.dir/CheckSymbolExists.c.o  -o cmTC_4cc55 -rdynamic 
08:12:37 CMakeFiles/cmTC_4cc55.dir/CheckSymbolExists.c.o: In function `main':
08:12:37 ./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/CheckSymbolExists.c:8: undefined reference to `pthread_create'
08:12:37 collect2: error: ld returned 1 exit status
08:12:37 CMakeFiles/cmTC_4cc55.dir/build.make:97: recipe for target 'cmTC_4cc55' failed
08:12:37 make[3]: *** [cmTC_4cc55] Error 1
08:12:37 make[3]: Leaving directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 Makefile:126: recipe for target 'cmTC_4cc55/fast' failed
08:12:37 make[2]: *** [cmTC_4cc55/fast] Error 2
08:12:37 make[2]: Leaving directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 
08:12:37 File /tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/CheckSymbolExists.c:
08:12:37 /* */
08:12:37 #include <pthread.h>
08:12:37 
08:12:37 int main(int argc, char** argv)
08:12:37 {
08:12:37   (void)argv;
08:12:37 #ifndef pthread_create
08:12:37   return ((int*)(&pthread_create))[argc];
08:12:37 #else
08:12:37   (void)argc;
08:12:37   return 0;
08:12:37 #endif
08:12:37 }
08:12:37 
08:12:37 Determining if the function pthread_create exists in the pthreads failed with the following output:
08:12:37 Change Dir: /tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp
08:12:37 
08:12:37 Run Build Command:"/usr/bin/make" "cmTC_75eb9/fast"
08:12:37 make[2]: Entering directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 /usr/bin/make -f CMakeFiles/cmTC_75eb9.dir/build.make CMakeFiles/cmTC_75eb9.dir/build
08:12:37 make[3]: Entering directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 Building C object CMakeFiles/cmTC_75eb9.dir/CheckFunctionExists.c.o
08:12:37 /usr/bin/cc    -g -O2 -fdebug-prefix-map=/tmp/binarydeb/ros-melodic-map-organizer-0.8.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DCHECK_FUNCTION_EXISTS=pthread_create   -o CMakeFiles/cmTC_75eb9.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.7/Modules/CheckFunctionExists.c
08:12:37 Linking C executable cmTC_75eb9
08:12:37 /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_75eb9.dir/link.txt --verbose=1
08:12:37 /usr/bin/cc  -g -O2 -fdebug-prefix-map=/tmp/binarydeb/ros-melodic-map-organizer-0.8.4=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -DCHECK_FUNCTION_EXISTS=pthread_create    CMakeFiles/cmTC_75eb9.dir/CheckFunctionExists.c.o  -o cmTC_75eb9 -rdynamic -lpthreads 
08:12:37 /usr/bin/ld: cannot find -lpthreads
08:12:37 collect2: error: ld returned 1 exit status
08:12:37 CMakeFiles/cmTC_75eb9.dir/build.make:97: recipe for target 'cmTC_75eb9' failed
08:12:37 make[3]: *** [cmTC_75eb9] Error 1
08:12:37 make[3]: Leaving directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 Makefile:126: recipe for target 'cmTC_75eb9/fast' failed
08:12:37 make[2]: *** [cmTC_75eb9/fast] Error 2
08:12:37 make[2]: Leaving directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4/obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp'
08:12:37 
08:12:37 
08:12:37 	cd /tmp/binarydeb/ros-melodic-map-organizer-0.8.4
08:12:37 dh_auto_configure: cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_BUILD_TYPE=None -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCATKIN_BUILD_BINARY_PACKAGE=1 -DCMAKE_INSTALL_PREFIX=/opt/ros/melodic -DCMAKE_PREFIX_PATH=/opt/ros/melodic returned exit code 1
08:12:37 debian/rules:25: recipe for target 'override_dh_auto_configure' failed
08:12:37 make[1]: Leaving directory '/tmp/binarydeb/ros-melodic-map-organizer-0.8.4'
08:12:37 make[1]: *** [override_dh_auto_configure] Error 255
08:12:37 debian/rules:22: recipe for target 'build' failed
08:12:37 make: *** [build] Error 2
08:12:37 dpkg-buildpackage: error: debian/rules build gave error exit status 2
08:12:37 E: Building failed
08:12:37 Traceback (most recent call last):
08:12:37   File "/tmp/ros_buildfarm/ros_buildfarm/binarydeb_job.py", line 138, in build_binarydeb
08:12:37     subprocess.check_call(cmd, cwd=source_dir)
08:12:37   File "/usr/lib/python3.5/subprocess.py", line 271, in check_call
08:12:37     raise CalledProcessError(retcode, cmd)
08:12:37 subprocess.CalledProcessError: Command '['apt-src', 'build', 'ros-melodic-map-organizer']' returned non-zero exit status 1
08:12:37 # END SUBSECTION
```

Some other packages are failing due to the same error.
http://build.ros.org/view/Mbin_ds_dS64/job/Mbin_ds_dS64__openrtm_ros_bridge__debian_stretch_amd64__binary/14/console
http://build.ros.org/view/Mbin_ds_dS64/job/Mbin_ds_dS64__hrpsys__debian_stretch_amd64__binary/22/console
http://build.ros.org/view/Mbin_ds_dS64/job/Mbin_ds_dS64__librealsense2__debian_stretch_amd64__binary/117/console

```
08:17:55 -- Configuring done
08:17:55 CMake Error at CMakeLists.txt:45 (add_executable):
08:17:55   Target "safety_limiter" links to target "Qt5::Widgets" but the target was
08:17:55   not found.  Perhaps a find_package() call is missing for an IMPORTED
08:17:55   target, or an ALIAS target is missing?
```
https://github.com/at-wat/neonavigation/pull/473 seems still needed.

--- a/map_organizer/CMakeLists.txt
+++ b/map_organizer/CMakeLists.txt
@@ -38,6 +38,16 @@ include_directories(${catkin_INCLUDE_DIRS} ${PCL_INCLUDE_DIRS} ${Eigen3_INCLUDE_
 add_definitions(${PCL_DEFINITIONS})
 
 
+# Workaround for debian stretch build (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=894656)
+list(REMOVE_ITEM PCL_LIBRARIES
+  "vtkGUISupportQt"
+  "vtkGUISupportQtOpenGL"
+  "vtkGUISupportQtSQL"
+  "vtkGUISupportQtWebkit"
+  "vtkViewsQt"
+  "vtkRenderingQt"
+)
+
 # Workaround for the bug in PCL(<1.8.1) https://github.com/PointCloudLibrary/pcl/issues/1406
 remove_definitions(-DDISABLE_LIBUSB-1.0)
 

--- a/obj_to_pointcloud/CMakeLists.txt
+++ b/obj_to_pointcloud/CMakeLists.txt
@@ -26,6 +26,16 @@ include_directories(${catkin_INCLUDE_DIRS} ${PCL_INCLUDE_DIRS} ${Eigen3_INCLUDE_
 add_definitions(${PCL_DEFINITIONS})
 
 
+# Workaround for debian stretch build (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=894656)
+list(REMOVE_ITEM PCL_LIBRARIES
+  "vtkGUISupportQt"
+  "vtkGUISupportQtOpenGL"
+  "vtkGUISupportQtSQL"
+  "vtkGUISupportQtWebkit"
+  "vtkViewsQt"
+  "vtkRenderingQt"
+)
+
 # Workaround for the bug in PCL(<1.8.1) https://github.com/PointCloudLibrary/pcl/issues/1406
 remove_definitions(-DDISABLE_LIBUSB-1.0)
 

--- a/safety_limiter/CMakeLists.txt
+++ b/safety_limiter/CMakeLists.txt
@@ -33,6 +33,16 @@ include_directories(${catkin_INCLUDE_DIRS} ${PCL_INCLUDE_DIRS} ${Eigen_INCLUDE_D
 add_definitions(${PCL_DEFINITIONS})
 
 
+# Workaround for debian stretch build (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=894656)
+list(REMOVE_ITEM PCL_LIBRARIES
+  "vtkGUISupportQt"
+  "vtkGUISupportQtOpenGL"
+  "vtkGUISupportQtSQL"
+  "vtkGUISupportQtWebkit"
+  "vtkViewsQt"
+  "vtkRenderingQt"
+)
+
 # Workaround for the bug in PCL(<1.8.1) https://github.com/PointCloudLibrary/pcl/issues/1406
 remove_definitions(-DDISABLE_LIBUSB-1.0)
 

